gammapy/modeling/selection.py

```diff
@@ -0,0 +1,105 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from gammapy.modeling import Fit, Parameter
+from gammapy.stats.utils import sigma_to_ts
+
+
+class TestStatisticNested:
+    """Compute the test statistic (TS) between two nested hypothesis .
+    The null hypothesis is the minimal one, for which a set of parameters
+    are frozen to given values. The model is updated to the alternative hypothesis
+    if there is a significant improvement (larger than the given threshold).
+
+    Parameters
+    ----------
+    parameters : `~gammapy.modeling.Parameters` or list of `~gammapy.modeling.Parameter`
+        List of parameters frozen for the null hypothesis but free for the test hypothesis.
+    null_values : list of float or `~gammapy.modeling.Parameters`
+        Values of the parameters frozen for the null hypothesis.
+        If a `Parameters` object or a list of `Parameters` is given
+        the null hypothesis follows the values of these parameters,
+        so this tests linked parameters versus unliked.
+    n_sigma : float
+        Threshold in number of sigma to switch from the null hypothesis
+        to the alternative one. Default is 2.
+    n_free_parameters : int
+        Number of free parameters to consider between the two hypothesis
+        in order to estimate the `ts_threshold` from the `n_sigma` threshold.
+        Default is len(parameters).
+    fit : `Fit`
+        Fit instance specifying the backend and fit options.
+    """
```

Comment:
the `n_sigma` to DeltaTS conversion supposes that the Wilk's theorem is valid for this dataset. Maybe one should precise this into the docstring (otherwise, as mentioned by @adonath, simulations should be done)

Reply:
added